_posts/2019-03-28-Add-Files-From-Git-Repos.md

```diff
@@ -0,0 +1,150 @@
+---
+layout: post
+title: Add Files From Git Repositories
+author: bcl
+tags: lorax-composer
+---
+
+When you are building an image sometimes you need to add some configuration
+files, extra tools, etc.  The temptation is to just copy them over from your
+local system and move on, but that isn't the right answer.  Unless you document
+where the files came from nobody looking at the image in 6 months will remember
+which version of them was included, where they came from, or even who created
+them.
+
+The right way to do this is to take your files and package them into an rpm,
+using the description to document their source, and the package version to keep
+track of them as you add them to an extra repository.  You can then add this
+repository to lorax-composer using the `composer-cli source` commands.
+
+Or, you could let composer handle creating the rpm for you and just point it
+to the git repository of files you want to include.
+
+In version 31.0 of lorax-composer we have added a new feature to the blueprints
+that does exactly that.  You add a section to the blueprint pointing to the git
+repo and composer will make an rpm of it and install the files under the
+destination path that you specify.  The rpm that is installed in the image
+includes the exact details about where the files came from so there is no
+question about their source.
+
+Let's walk through an example scenario with nginx and some virtual hosts.
+
+Nginx virtual hosts
+-------------------
+
+You can easily setup virtual hosts in nginx by adding configuration files to the
+`/etc/nginx/conf.d/` directory.  In their simplest form they look like this:
+
+    server {
+            listen 80;
+            server_name one-site.myhost.com;
+            root /var/www/one-site.myhost.com/public_html;
+            index index.php index.html;
+    }
+
+I have setup an [example git repository here](https://github.com/bcl/extra-files), it includes 3 vhosts
```

Comment:
Minor typo here :-)
Should be the verb "set up" and not the noun "setup". Also above at line 35.

Reply:
I do that alot :) (a joke, I also misuse a lot all the time). Thanks!